app/controllers/authenticate_controller.rb

```diff
@@ -223,6 +231,35 @@ def handle_authentication_error(err)
     end
   end
 
+  def audit_authn_error(err)
+    authenticate_audit_event(err)
+  end
+
+  def audit_success
+    Audit.logger.log(
+      authenticate_audit_event(nil)
+    )
+  end
+
+  def authenticate_audit_event(err)
```

Comment:
let's make this more generic as we call this also from `validate_status`
```suggestion
  def audit_failure(err)
```